### ISSUE TITLE
Unable to build node-sqlite3 in Synology NAS (arm-none-linux-gnueabi)

### ISSUE THREAD

**Issue by sromeroi** [mapbox/node-sqlite3#457]
Hi,

I'm trying to install sqlite3 in my Synology NAS and I get the following error:

```
sromero@NAS:~/test $ npm install sqlite3 --build-from-source=sqlite3     
> sqlite3@3.0.8 install /volume1/homes/sromero/test/node_modules/sqlite3
> node-pre-gyp install --fallback-to-build

child_process: customFds option is deprecated, use stdio instead.
child_process: customFds option is deprecated, use stdio instead.
make: Entering directory `/volume1/homes/sromero/test/node_modules/sqlite3/build'
Makefile:1: /env.mak: No such file or directory
make: *** No rule to make target `/env.mak'.  Stop.

make: Leaving directory `/volume1/homes/sromero/test/node_modules/sqlite3/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1067:12)
gyp ERR! System Linux 3.2.40
(etc...)
```

I've installed autoconf, automake, m4, make, gcc ...:

```
sromero@NAS:~ $ make -v
GNU Make 3.82
Built for arm-none-linux-gnueabi
Copyright (C) 2010  Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

sromero@NAS:~ $ cc -v
Using built-in specs.
Target: arm-none-linux-gnueabi
Configured with: ../gcc-4.2/configure --build=i386-pc-linux-gnu --host=arm-none-linux-gnueabi
--target=arm-none-linux-gnueabi --prefix=/opt --disable-nls --disable-static 
(...)
Thread model: posix
gcc version 4.2.3

sromero@NAS:~ $ uname -a
Linux NAS 3.2.40 #5565 SMP Mon May 11 02:13:03 CST 2015 armv7l GNU/Linux synology_armadaxp_ds214+
```

I found another issue similar (https://github.com/mapbox/node-sqlite3/issues/368) and I already applied the symlink fix (to have a valid cc binary) and it's still not compiling...

Any help will be appreciated, as I really need node-sqlite3 in my NAS for a node application ...

Thanks.

**Comment by springmeyer** [comment]
> Makefile:1: /env.mak: No such file or directory

That is the error I think you should focus on. However, I have no idea what would cause it. Never seen this before with node-sqlite3. Based on a quick search (https://www.google.com/webhp?sourceid=chrome-instant&ion=1&espv=2&ie=UTF-8#q=env.mak%3A+No+such+file+or+directory) it seems to be happening on only Synology systems. So I presume that something about `make` on Synology is fundamentally broken and that this is not fixable by doing anything different with node-sqlite3.

**Comment by sromeroi** [comment]
Thanks for your really quick answer.

Yes, you're right, it seems a problem with the build environment. I was asking here hoping that you any user already faced this problem and had a solution.

I'll ask in Synology's forums. 

Thanks!

**Comment by springmeyer** [comment]
Okay, I will close this then. Re-open a new issue if you are still having trouble.

**Comment by HooK81** [comment]
Hello !
I know this post is closed but anyone has found a solution ?

I try to install node sqlite3 in my Synology since 2 days without success.

I have the same error explained by sromeroi...

Thanks

**Comment by sromeroi** [comment]
Sorry, I didn't see your question, HooK81.

I went to Synology forums and got no valid solution, so I finally gave up trying to compile it..

I ended by using a Banana Pi (1GB RAM, Dual Core, EXTERNAL -no USB- SATA DISK accepted!!!) with Bananian Linux as the webserver (nodejs, sqlite, nginx, etc) mounting the desdired NAS directory via NFS ... This way I can install whatever I want and this has additional uses (webserver, mailserver, DNS, etc) letting the nas "more free" for doing NAS-things :). And now I don't need most of the custom IPKG packages I installed on the NAS, so I can safely "reinstall" synology software and don't miss anything. 

Hope this idea (using a RaspberryPi or BananaPi) helps....
